docs/general/ad-filtering/create-own-filters.md

```diff
@@ -1737,6 +1735,182 @@ As a response to blocked request AdGuard returns a short video placeholder.
 >
 > Rules with `$mp4` modifier are not supported by AdGuard Content Blocker, AdGuard for iOS and Safari.
 
+### Rule priorities {#rule-priorities}
+Each rule has its own priority, which is necessary when several rules match the request and the filtering system needs to select one of them.
+Priority is measured by a natural integer. The higher the priority of a rule, the higher its priority value.
+In the case of a conflict between two rules with the same priority value, there is undefined behavior, depending on the implementation.
+
+#### Priority counting
+
+To calculate priority, modifiers have been divided into several groups. They are arranged in order of priority from lowest to highest. The more a modifier narrows the scope of a rule, the more priority it adds to the total weight of the rule or the more a rule covers requests, the lower its priority. Note that there are modifiers where setting one parameter has a higher priority than setting two or more, for example, for `$domain=example.com|example.org`, because a rule with two domains has a smaller, infinitesimal, effective area, but still larger than a rule with one resolved domain.
+
+The base priority weight of any rule is 1. If the priority is a floating point number, it will be rounded to the **nearest whole number**.
+
+> **Compatibility with different versions of AdGuard**
+>
+> Numerical priorities for rules are available in tsurlfilter since version `v2.1.0`.
+
+
+#### 1. Basic modifiers, the presence of each adds 1 to the priority: {#priority-category-1}
+ * [`$third-party`](#third-party-modifier),
+ * [`$match-case`](#match-case-modifier),
+ * [`$to`](#to-modifier),
+ * [`$method`](#method-modifier),
+ * [`$denyallow`](#denyallow-modifier),
+ * [`$header`](#header-modifier),
+ * [`$dnsrewrite`](#dnsrewrite-modifier),
+ * [`$app`](#app-modifier) with forbidden applications using `~`,
+ * [`$domain`](#domain-modifier) with forbidden domains using `~`,
+ * banned content-types with `~`.
+
+For forbidden `$domain` and content-type, we add 1 for the presence of the modifier itself disregarding the number of forbidden domains or content-types, because the scope of the rule is infinitely large in the limit anyway, in other words, by banning several domains and content-types, we only *insignificantly* narrowed the scope of the rule.
```

Comment:
> For forbidden `$domain` and content-type, we add 1 for the presence of the modifier itself disregarding the number of forbidden domains or content-types
we add 1 what?

because the scope of the rule is infinitely large in the limit anyway —> because the scope of the rule is infinitely large anyway

, in other words, by banning several domains and content-types, we only *insignificantly* narrowed the scope of the rule. — consider making it a separate sentence